tests/model_serving/model_server/serverless/utils.py

```diff
@@ -50,6 +51,56 @@ def verify_no_inference_pods(client: DynamicClient, isvc: InferenceService) -> N
         raise
 
 
+def wait_for_deployments(
```

Comment:
you can use `@retry` decorator from [timeout sampler](https://github.com/RedHatQE/timeout-sampler) here. It would be simpler.